src/components/POLine/Location/LocationForm.js

```diff
@@ -16,81 +16,98 @@ import {
-    component={RepeatableField}
-    name="locations"
-    renderField={(field) => (
-      <React.Fragment>
-        <Col xs={6}>
-          <Field
-            component={Select}
-            dataOptions={getLocationsForSelect(parentResources)}
-            fullWidth
-            label={<FormattedMessage id="ui-orders.location.nameCode" />}
-            name={`${field}.locationId`}
-            placeholder=" "
-            required
-            validate={required}
-          />
-        </Col>
-        <Col xs={3}>
-          <Field
-            component={TextField}
-            label={<FormattedMessage id="ui-orders.location.quantityPhysical" />}
-            name={`${field}.quantityPhysical`}
-            parse={parseQuantity}
-            type="number"
-            validate={validateQuantityPhysical}
-          />
-        </Col>
-        <Col xs={3}>
-          <Field
-            component={TextField}
-            label={<FormattedMessage id="ui-orders.location.quantityElectronic" />}
-            name={`${field}.quantityElectronic`}
-            parse={parseQuantity}
-            type="number"
-            validate={validateQuantityElectronic}
-          />
-        </Col>
-      </React.Fragment>
-    )}
-  />
-);
+const LocationForm = ({ parentResources, orderFormat }) => {
+  const isNeedElValidation = ERESOURCES.includes(orderFormat);
+  const isNeedPhValidation = [...PHRESOURCES, OTHER].includes(orderFormat);
+
+  return (
+    <FieldArray
+      addLabel={<FormattedMessage id="ui-orders.location.button.addLocation" />}
+      component={RepeatableField}
+      name="locations"
+      renderField={(field) => (
+        <React.Fragment>
+          <Col xs={6}>
+            <Field
+              component={Select}
+              dataOptions={getLocationsForSelect(parentResources)}
+              fullWidth
+              label={<FormattedMessage id="ui-orders.location.nameCode" />}
+              name={`${field}.locationId`}
+              placeholder=" "
+              required
+              validate={required}
+            />
+          </Col>
+          <Col xs={3}>
+            <Field
+              component={TextField}
+              label={<FormattedMessage id="ui-orders.location.quantityPhysical" />}
+              name={`${field}.quantityPhysical`}
+              parse={parseQuantity}
+              type="number"
+              disabled={!isNeedPhValidation}
+              validate={isNeedPhValidation ? [validateNotNegative, validateQuantityPhysical] : noop}
+            />
+          </Col>
+          <Col xs={3}>
+            <Field
+              component={TextField}
+              label={<FormattedMessage id="ui-orders.location.quantityElectronic" />}
+              name={`${field}.quantityElectronic`}
+              parse={parseQuantity}
+              type="number"
+              disabled={!isNeedElValidation}
+              validate={isNeedElValidation ? [validateNotNegative, validateQuantityElectronic] : noop}
```

Comment:
what do you think of leaving origin code? it worked as expected and tested - just to check if numbers of quantities matches. It was tested, at least manually

Reply:
related with disabled property we need to prevent enter quantity in field doesn't related to the order format, as you know I've harried to ask Dennis about not negative values for quantity and now have to check it. Any case I need to include this check to code in separate method or I'll need to extends existing methods. I'd appreciate to get you vision on this issue

Reply:
yep, as we agreed I left original code, but added not negative validation in separate method, as @NikitaSedyx mentioned there was a copy&past